cloudbuild-prod.yaml

```diff
@@ -1,23 +1,49 @@
 # Empty
 steps:
   - name: 'gcr.io/cloud-builders/npm'
+    id: install_frontend_dep
     args: ['install']
+
   - name: 'gcr.io/cloud-builders/npm'
+    id: install_functions_dep
+    args: ['run', 'installFunctions']
+    waitFor: ['-']
+
+  - name: 'gcr.io/$PROJECT_ID/firebase'
+    entrypoint: bash
+    args:
+      [
+        '-c',
+        'firebase deploy --only functions --project $PROJECT_ID --token $$ENC_FIREBASE_TOKEN',
+      ]
+    secretEnv: ['ENC_FIREBASE_TOKEN']
+    waitFor: ['install_functions_dep']
+
+  - name: 'gcr.io/cloud-builders/npm'
+    id: gatsby_build
     args: ['run', 'build']
     env:
       [
         'GATSBY_HANDLE_APPLICATION_ENDPOINT=https://europe-west1-dps-website-244212.cloudfunctions.net/handleApplicationForm',
         'CURR_ENV=production',
       ]
+    waitFor: ['install_frontend_dep']
+
   - name: 'gcr.io/cloud-builders/gcloud'
     args:
       [
         'app',
         'deploy',
         'website-prod.app.yaml',
         '--project',
-        'dps-website-244212',
+        '$PROJECT_ID',
         '--log-http',
         '--verbosity',
         'debug',
       ]
+    waitFor: ['gatsby_build']
+
+secrets:
+  - kmsKeyName: projects/dps-website-244212/locations/global/keyRings/website-staging/cryptoKeys/website-staging-deploykey
```

Comment:
This staging in the key name is just a typo or do we really use something from staging here?

Reply:
Just a typo actually and tried to change it but couldn't :/